src/codegen/breakout.rs

```diff
@@ -106,14 +106,15 @@ impl<'a> BreakoutTarget<'a> {
     where
         'a: 'b,
     {
+        b.position_at_end(self.bb);
```

Comment:
As I understand it, it has to do with how a call to `compile_block` to generate a "WebAssembly block" can generate multiple LLVM BBs due to internal control flow within that "WebAssembly block." When this occurs, the LLVM builder writes the LLVM BBs in order, and by the time it reaches `build_result`, it is positioned at the bottom of the last BB (potentially after a branch and thus unreachable). We seem to hit this as the result of condition breaks (br_if, br_table, etc.), because these generate subsequent BBs for the "else" case. I've attached the wat and bitcode for the tests that fail when this is uncommented. My impression is that in deeply nested control flow (like with float parsing in libc), the resulting bitcode actually passes validation, but the actual control flow is still impacted. I don't really grok how that happens, but float parsing works properly after this fix. 
[control_flow_bug_examples.zip](https://github.com/gwsystems/aWsm/files/7461446/control_flow_bug_examples.zip)